### ISSUE TITLE
Enhancement: $lastpost_subject > 25

### ISSUE THREAD

**Issue by woostarUK** [mybb/mybb#3565]
The number of characters shouldn't be hard coded. It should be set in Forumhome or Forumdisplay settings.

Different templates, not to mention mobile screens, require this to be adaptable.

inc/functions_forumlist.php
Line 312
```
if(my_strlen($lastpost_subject) > 25)
{
	$lastpost_subject = my_substr($lastpost_subject, 0, 25)."...";
}
```

----------

Edit: Infact, can there be no hardcoded limit and use css text-overflow: ellipsis; in the template?

**Comment by Eldenroot** [comment]
I like the idea with CSS :)

**Comment by Sama34** [comment]
Yeah, I think @justinsoltesz mentioned about using CSS for later versions.

**Comment by euantorano** [comment]
I wouldn't mind using a CSS solution in 1.8, and I believe it's already done in 1.9?

**Comment by justinsoltesz** [comment]
I don’t think it’s done in 1.9 yet but certainly can be.

I **strongly discourage** making CSS changes in 1.8 because these are not automatically upgraded and are not tracked at all. This has been an issue several times with previous 1.8.x releases.

**Comment by euantorano** [comment]
Ok, sounds good to me then. I momentarily forgot about our CSS issues on upgrade.

**Comment by justinsoltesz** [comment]
I think it’s a good reminder for all as it has certainly been forgotten or overlooked in the past!

**Comment by Eldenroot** [comment]
Yeah, so the only way it to change hardcoded value into the setting (in 1.8.x branch)

**Comment by yuliu** [comment]
I vote for a CSS or JavaScript solution, or there should be additional settings for title cutoff length in _Forum Home Options_ and _Forum Display Options_ and that isn't adaptable for different themes.

Since there's upgrade issue for 1.8, I suggest making changes in 1.9 if necessary.

**Comment by Sama34** [comment]
Hi, I think avoiding CSS changes is sound reasoning, but I think that CSS changes should be acceptable as nothing breaks if administrators don't apply such changes.

For this specific case, adding a CSS solution and use `{$full_lastpost_subject}` instead of `{$lastpost_subject}` in the default theme should be fine.

**Comment by dvz** [comment]
> I think that CSS changes should be acceptable as nothing breaks if administrators don't apply such changes.

It looks like it does - the old variable may indeed be left in themes that have the relevant template(s) customized, but otherwise the server-side trimming is removed while the CSS has to be applied manually.

IMO this should be reverted and deferred to 1.9.

**Comment by Sama34** [comment]
But server-side trimming wasn't removed with the PR for this #4344 , so highlighting this change in the release post was the only left step.
https://github.com/mybb/mybb/blob/4817069fcaa5a5fcb82e34fc7f35091810c347e2/inc/functions_forumlist.php#L313-L318

**Comment by dvz** [comment]
> highlighting this change in the release post was the only left step

Yes, the PR seems OK; the question is whether it would be worth forcing administrators to perform the extra step of updating CSS (otherwise the table layout may break when templates change automatically), given that it's a barely noticeable visual improvement which can be addressed in 1.9.

**Comment by Eldenroot** [comment]
This is a really stupid decision... srry to say!

**Comment by Stefan-MyBB** [comment]
Updating the CSS (and potentially breaking many themes) on the last release of a series to make a small improvement isn't very smart either.

**Comment by Eldenroot** [comment]
Guys, I dont get this... there were other CSS changes in 1827, these requires manually addition into global.css.

For example this one: 

https://github.com/mybb/mybb/issues/4136

**Comment by dvz** [comment]
Those changes could be skipped without breaking anything: https://github.com/mybb/mybb/pull/4301#discussion_r611735295

**Comment by arcalys** [comment]
The final decision is unclear. Can you please confirm what change is expected?

**Comment by dvz** [comment]
The PHP trimming should be replaced with CSS `text-overflow: ellipsis` for e.g. `.forum__subject` in 1.9.

It looks like there were issues when `width` was not specified, but it may work with e.g. `overflow: hidden; white-space: nowrap;`.

There are a few similar calls in other files (`portal.php`, `modcp.php`, `usercp.php`) that may be easily solved, depending on the templates.

**Comment by Sama34** [comment]
This should be done in 1.9 and close this as rejected for 1.8

**Comment by dvz** [comment]
We can keep it for 1.9 - the label was swapped already
